src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -820,6 +978,39 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
             }
         }
 
+        private static readonly DiagnosticBag s_notUsedDiagnosticBagForDerivingArgumentsForIOperation = new DiagnosticBag();
```

Comment:
Please consider using a local `DiagnosticBag`, allocated from the pool, in the method below instead.